### PR TITLE
metrics service: allow only histogram summaries

### PR DESCRIPTION
**Pull request by sschepens**
Commit Message: Allow only histogram summaries in metric service
Additional Description: Histogram metrics by default send two metrics Summary (containing quantiles) and Histogram (containing all the buckets), this makes payloads extremely large when handling lots of stats. In some cases, only having the summary is enough, since we can get sum, count, and percentiles. Furthermore, it probably only makes sense to send one of both, if I were to need buckets, then Summary is probably useless.
Risk Level: low, off by default
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/envoy/config/metrics/v3/metrics_service.proto
Comment (jmarantz) [inline api/envoy/config/metrics/v3/metrics_service.proto:62]: In the PR description you described that if you actually wanted the buckets then you wouldn't want the summary.

Should we add here the ability to send all combinations of buckets, summaries, or nothing?

**Review by jmarantz — COMMENTED** [review]
should we have some tests for this new mode?

**Comment by sschepens** [comment]
> should we have some tests for this new mode?

yes! on it

**Comment by sschepens** [comment]
@jmarantz pushed a new implementation using an enum to allow specifying which metrics you want emitted, also added tests for this.
The only thing i'm not sure is if protobuf defaults to the enum value `0` when not specified or if we need some other sort of default.

File: api/envoy/config/metrics/v3/metrics_service.proto
Comment (ramaraochavali) [inline api/envoy/config/metrics/v3/metrics_service.proto:73]: nit:` histogram_emit_mode` may be better name?

Comment (sschepens) [reply api/envoy/config/metrics/v3/metrics_service.proto:73]: sure

Comment (sschepens) [reply api/envoy/config/metrics/v3/metrics_service.proto]: @ramaraochavali done

**Review by mattklein123 — CHANGES_REQUESTED** [review]
API LGTM. Please add a release note. I will defer to @jmarantz for code review. Thank you!

**Comment by alyssawilk** [comment]
@jmarantz ping!

**Comment by jmarantz** [comment]
many apologies for the delay in responsiveness on this and other PRs; I will try to catch up over the weekend.

File: test/extensions/stats_sinks/metrics_service/grpc_metrics_service_impl_test.cc
Comment (jmarantz) [inline test/extensions/stats_sinks/metrics_service/grpc_metrics_service_impl_test.cc:335]: nit: just break these up 3 three separate test methods. You already have the common prep factored to one nice helper function.

File: test/extensions/stats_sinks/metrics_service/grpc_metrics_service_impl_test.cc
Comment (jmarantz) [inline test/extensions/stats_sinks/metrics_service/grpc_metrics_service_impl_test.cc:348]: s/ASSERT(/EXPECT_TRUE(/ here and below.

ASSERT will crash the test binary if it fails. ASSERT_TRUE will fail the test and early-exit the method if it fails,  and EXPECT_TRUE will fail the test and continue. I think EXPECT_TRUE is what you want here.

**Review by jmarantz — COMMENTED** [review]
LGTM modulo test nits. Thank you and sorry for the delay.

**Review by jmarantz — APPROVED** [review]
Thank you!

**Comment by jmarantz** [comment]
Can you merge main? I think that might fix the 'dependencies' test failure.

**Comment by sschepens** [comment]
added changelog and merged main

**Review by jmarantz — DISMISSED** [review]
I think you need to merge main again to get rid of the spurious arm64 CI failure, which was a flake I think.

**Review by lizan — DISMISSED** [review]
can you resolve conflicts? defer to @jmarantz to merge.

**Comment by jmarantz** [comment]
@sschepens  you might be interested in this related issue: https://github.com/envoyproxy/envoy/issues/11259

I have a skeletal draft PR https://github.com/envoyproxy/envoy/pull/26472 which exposes raw histogram bins as an admin endpoint via json. It might be interesting to also access those via metrics service API, which would add a new enum choice to what you have here.

**Review by jmarantz — APPROVED** [review]

**Pull request merged**
